[IMP] accounting/l10n_br: electronic invoicing & tax computation flow (#8784)
odoo/documentation: content/applications/finance/fiscal_localizations/brazil.rst

Configuration
=============

Modules installation
--------------------

:ref:`Install <general/install>` the following modules to get all the features of the Brazilian
localization:

.. list-table::
   :header-rows: 1
   :widths: 25 25 50

   * - Name
     - Technical name
     - Description
   * - :guilabel:`Brazilian - Accounting`
     - `l10n_br`
     - Default :ref:`fiscal localization package <fiscal_localizations/packages>`, which represents
       having the Generic Brazilian chart of accounts and Taxes, together with document types and
       identification types.
   * - :guilabel:`Avatax Brazil & Avatax Brazil for Services`
     - `l10n_br_avatax & l10n_br_avatax_services`
     - Goods and Services tax computation through Avalara.

>    * - :guilabel:`Brazilian Accounting EDI & Avatax Brazilian Accounting EDI for services`

Reviewer: Thanks @dade-odoo, yes it appears Brazil Pix QR codes (l10n_br_pix) does not exist on saas-17.2. @vbe-odoo, can you confirm?

If so, I will plan to remove lines 54-56 on the saas-17.2 FWP.

>    * - :guilabel:`Brazilian Accounting EDI & Avatax Brazilian Accounting EDI for services`

Author: Yup, you are right, I can see that l10n_br_pix was merged in 17.2 into l10n_br (https://github.com/odoo/odoo/pull/157324)

@samueljlieber can you please help me remove those lines (54-56 on the saas-17.2 FWP)?

Thanks!

#. Select the :guilabel:`Company` option at the top of the page. Then, configure the following
   fields:

   - :guilabel:`Name`
   - :guilabel:`Address` (add :guilabel:`City`, :guilabel:`State`, :guilabel:`Zip Code`,
     :guilabel:`Country`)

     - In the :guilabel:`Street` field, enter the street name, number, and any additional address
       information.
     - In the :guilabel:`Street 2` field, enter the neighborhood.

>    - :guilabel:`Identification Number` (:guilabel:`CNPJ`, :guilabel:`CPF`)

Reviewer 2: I would put the description or options after a semicolon instead of between brackets (it does't bother me that much here, but it does in other similar lists elsewhere in this doc so for the sake of consistency, I'd have them all the same)

   - :guilabel:`Tax ID` (associated with the identification type)
   - :guilabel:`IE` (State registration)
   - :guilabel:`IM` (Municipal registration)
   - :guilabel:`SUFRAMA code` (Superintendence of the Manaus Free Trade Zone - add if applicable)
   - :guilabel:`Phone`
   - :guilabel:`Email`

   .. image:: brazil/contact-configuration.png
      :alt: Company configuration.

#. Configure the :guilabel:`Fiscal Information` within the :guilabel:`Sales and Purchase` tab:

   - Add the :guilabel:`Fiscal Position` for :ref:`Avatax Brazil <brazil/fiscal-positions>`.

>    - :guilabel:`Tax Regime` (Federal Tax Regime)
>    - :guilabel:`ICMS Taxpayer Type` (indicates ICMS regime, Exempt status, or Non-Taxpayer)

Reviewer 2: ```suggestion
   - :guilabel:`Tax Regime`: Federal Tax Regime
   - :guilabel:`ICMS Taxpayer Type`: Indicates :guilabel:`ICMS regime`, :guilabel:`Exempt status`, or :guilabel:`Non-Taxpayer`
```

   - :guilabel:`Main Activity Sector`

   .. image:: brazil/contact-fiscal-configuration.png
      :alt: Company fiscal configuration.

#. Configure the following extra :guilabel:`Fiscal Information` if you are going to issue NFS-e:

   - Add the :guilabel:`Fiscal Position` for :ref:`Avatax Brazil <brazil/fiscal-positions>`.

>    - :guilabel:`COFINS Details` (Taxable, Not Taxable, Taxable with rate 0%, Exempt, Suspended)
>    - :guilabel:`PIS Details` (Taxable, Not Taxable, Taxable with rate 0%, Exempt, Suspended)
>    - :guilabel:`CSLL Taxable` (If the company is subject to CSLL or not)

Reviewer 2: ```suggestion
   - :guilabel:`COFINS Details`: :guilabel:`Taxable, Not Taxable, Taxable with rate 0%, Exempt, Suspended`
   - :guilabel:`PIS Details`: :guilabel:`Taxable, Not Taxable, Taxable with rate 0%, Exempt, Suspended`
   - :guilabel:`CSLL Taxable`: If the company is subject to CSLL or not
```

   .. image:: brazil/contact-fiscal-configuration.png
      :alt: Contact fiscal configuration.

#. Configure the following extra :guilabel:`Fiscal Information` if you are going to issue NFS-e:

>    - Add the :guilabel:`Fiscal Position` for :ref:`Avatax Brazil <brazil/fiscal-positions>`

Reviewer 2: Same as above regarding the brackets + add GUI labels where needed (see my other comment on the same topic above)

Customer invoices
~~~~~~~~~~~~~~~~~

To process an electronic invoice for goods (NF-e) or services (NFS-e), the invoice needs to be
confirmed and taxes need to be computed by Avalara. Once that step is done, click on the

> :guilabel:`Send & Print` button in the upper left corner, and a pop-up will appear. Then click on

Reviewer 2: ```suggestion
:guilabel:`Send & Print` button in the upper left corner. In the pop-up that appears, click on
```

:guilabel:`Process e-invoice` and any of the other options - :guilabel:`Download` or
:guilabel:`Email`. Finally, click on :guilabel:`Send & Print` to process the invoice with the
government.

Before sending the electronic invoice for goods (NF-e) or services (NFS-e), some fields need to be
filled out on the invoice:

> - :guilabel:`Customer` with all the customer information

Reviewer 2: ```suggestion
- :guilabel:`Customer`, with all the customer information
```

> - :guilabel:`Payment Method: Brazil` (how the invoice is planned to be paid)

Reviewer 2: ```suggestion
- :guilabel:`Payment Method: Brazil`: how the invoice is planned to be paid
```

   Check whether the electronic invoice is still within the cancellation deadline, which may vary
   according to the legislation of each state.

> For goods e-invoices (NF-e)

Reviewer 2: ```suggestion
E-invoices for goods (NF-e)
```

***************************

Cancel an e-invoice for goods (NF-e) in Odoo by clicking :guilabel:`Request Cancel` and adding a
cancellation :guilabel:`Reason` on the pop-up that appears. If you want to send this cancellation
reason to the customer via email, activate the :guilabel:`E-mail` checkbox.

This is an electronic cancellation, which means that Odoo will send a request to the government to
cancel the NF-e, and it will then consume one |IAP| credit, as an |API| call occurs.

.. image:: brazil/invoice-cancellation.png
   :alt: Invoice cancellation reason in Odoo.

> For services e-invoices (NFS-e)

Reviewer 2: ```suggestion
E-invoices for services (NFS-e)
```

.. image:: brazil/correction-letter.png
   :alt: Correction letter reason in Odoo.

.. important::

>    This electronic document is only available for NF-e.

Reviewer 3: Specify which document exactly (for example, "Correction letters are only available for electronic invoices for goods (NF-e)."

.. image:: brazil/range-number-invalidation-wizard.png
   :alt: Number range invalidation wizard in Odoo.

.. important::

>    This electronic document is only available for NF-e.

Reviewer 3: Specify which document exactly (for example, "Invalidate invoice number range documents are only available for electronic invoices for goods (NF-e)."
